File: code/shawn/lab20-credit_card_validation/credit_card_validation.py
# Shawn Stolsig
# PDX Code Guild 
# Assignment: Lab 20 - Credit Card Validation
# Date: 10/23/2019

# sample number from lab assignment
# sampleNum = str(4556737586899855)
sampleNum = input("please input a sample number: ")

# define function for validating credit card info is correct
def checkCard(numStr):

    # 1. turn input string into list of numbers
    numList = [int(i) for i in numStr]
    print(numList)

    # 2. Slice off the last digit. That is the check digit.
    checkDigit = numList.pop()
    print(numList)

    # 3. Reverse the digits.
    numList.reverse()
    print(numList)

    # 4. Double every other element in the reversed list.
    numList = [numList[i]*2 if i % 2 == 0 else numList[i] for i in range(0,len(numList))]
    print(numList)

    # 5. Subtract nine from numbers over nine.
    numList = [num-9 if num > 9 else num for num in numList]
    print(numList)

    # 6. Sum all values.
    sum = 0
    for i in range(len(numList)):
        sum += numList[i]
   
    # 7. Take the second digit of that sum
    print(f"check digit is {checkDigit}")
    print(f"sum is {sum}")
    # 8. If that matches the check digit, the whole card number is valid.
    return str(checkDigit) == str(sum)[1]

if checkCard(sampleNum):
    print("Valid!")
else:
    print("Not valid :(")

File: code/rhi/pick3-02/credit_card_validation.js
// credit card validator
//  part one of the pick three javascript lab
//  written by Rhornberger
// last updated dec 16,2019

// turning my string of numbers into a list of intergers 
let u_num = prompt('Please enter a credit card number: ').split(' ').map(Number);
// alert(u_num);

// slice the last number off my list
let u_num2 = u_num.pop();
// alert(u_num2)

// turn the rest of the array around, the ending number is now the first number
let cc_num = u_num.slice(0,15);
cc_num1 = cc_num.reverse();
// alert(cc_num1)

// double every other item in the list
// let res1 = []
let res = cc_num1.map(function(val, i) {
    if(i % 2 === 0) {
        return val * 2
    } else {
        return val
    }
});
// alert(res)

// subtract 9 from all number greater than 9
let res2 = res.map(function(val) {
    if (val > 9) {
        return val - 9
    } else {
        return val
    }
});
// alert(res2)

// sum of all the values in the array
let tot = 0
let user_list = res2.map(function(num) {
    tot = tot + num;
    return tot
});
// alert(tot)

// get the second digit of the returned tot number
let tot2 = Math.floor(tot % 10);
// alert(tot2)

// compare the number from tot2 to the final number that was popped off in the beggining.
if (tot2 === u_num2) {
    alert('Congradulations! This is a valid credit card number.')
} else {
    alert(`I am sorry ${u_num} is not a valid credit card number.`)
}